### ISSUE TITLE
Paper notes: Attention Is All You Need

### ISSUE THREAD

**Issue by HeziLiu** [mcpmark-eval-liuhezi/transformer-study#2]
# Paper notes: Attention Is All You Need

**Paper:** Vaswani et al., "Attention Is All You Need" (arXiv:1706.03762)

## Outline (all section headings, exactly as in the paper, top-level and nested, in order)

1. Introduction
2. Background
3. Model Architecture
   - 3.1 Encoder and Decoder Stacks
   - 3.2 Attention
     - 3.2.1 Scaled Dot-Product Attention
     - 3.2.2 Multi-Head Attention
     - 3.2.3 Applications of Attention in our Model
   - 3.3 Position-wise Feed-Forward Networks
   - 3.4 Embeddings and Softmax
   - 3.5 Positional Encoding
4. Why Self-Attention
5. Training
   - 5.1 Training Data and Batching
   - 5.2 Hardware and Schedule
   - 5.3 Optimizer
   - 5.4 Regularization
6. Results
   - 6.1 Machine Translation
   - 6.2 Model Variations
   - 6.3 English Constituency Parsing
7. Conclusion
8. Attention Visualizations

## Key results

- **WMT 2014 English-to-German translation task:** "28.4 BLEU" — "Our model achieves 28.4 BLEU on the WMT 2014 English-to-German translation task, improving over the existing best results, including ensembles, by over 2 BLEU."
- **WMT 2014 English-to-French translation task:** "41.8" — "On the WMT 2014 English-to-French translation task, our model establishes a new single-model state-of-the-art BLEU score of 41.8 after training for 3.5 days on eight GPUs, a small fraction of the training costs of the best models from the literature."
